model/src/lang/reification.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::type_complexity)] // TODO: This is a temporary solution to avoid the warning. Fix type complexity.
```

Comment:
Not sure it is still required. Even if it is, it should be made more targeted (i.e. not file wide).

Reply:
Yes, I have reverted the changes.